### PR TITLE
Add lwip doc.

### PR DESCRIPTION
**Pull request by cecille**
#### Problem
There was a request to add the lwip requirements / improvements doc to the codebase to make discussion easier.

#### Change overview
Adds doc.

#### Testing
No tests - just a doc.

### CODE REVIEW

File: docs/discussion/lwip_ipv6.md [hunk 1]
```diff
@@ -0,0 +1,139 @@
+# LwIP changes for CHIP
+
+LwIP is the network layer used in the CHIP platform. Although it has some good
+IPv6 support, there are areas that are lacking that we should implement for
+CHIP. The recommendations here are listed roughly from most to least important.
+
+## Route Information Options (RIO)
+
+The addressing proposal requires devices to store route options from Route
```
Comment (tcarmelveilleux) [inline docs/discussion/lwip_ipv6.md:9]:
```suggestion
The specification requires devices to store route options from Route
```

File: docs/discussion/lwip_ipv6.md [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+Information Options (RIO) sent in router advertisements. This functionality is
+not currently present in upstream LwIP. The patch to add this is relatively
+small, but we may need to upstream this in order to require its use in CHIP. It
+can be added to the CHIP version, but not all platforms use the same middleware.
```
Comment (tcarmelveilleux) [inline docs/discussion/lwip_ipv6.md:13]: The CHIP version is used by none of the middleware IIUC, so we don't have real tests against changes of it.

File: docs/discussion/lwip_ipv6.md [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+
+### Recommendation:
+
+-   write a RIO patch, add to CHIP middleware, upstream to lwip
+    -   Ensure patch is RFC compliant (especially re: expiry)
+
+## DNS
+
+LwIP’s DNS handling isn’t great and breaks down when the router supports
```
Comment (tcarmelveilleux) [inline docs/discussion/lwip_ipv6.md:22]: CHIP doesn't require DNS support, except for time sync NTP, and optional HTTPS support for OTA download, both of which are optional in spec and likely not supported in the very short term roadmap. I would consider this to be a low priority item and we should move it to the bottom of the file/list to make it more obvious it's not a top item.

File: docs/discussion/lwip_ipv6.md [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+layer.
+
+Upstream LwIP has support for IPv6 address scopes, but only as an option.
+However, the code to support this is not present in the CHIP LwIP codebase.
+Other platform versions assume this option is not present (ex. M5 has an
+assertion on ip address sizes that disallow the use of a scope tag).
+
+### Recommendation:
+
+-   Update CHIP lwip version for scopes support, ensure CHIP code works with
```
Comment (tcarmelveilleux) [inline docs/discussion/lwip_ipv6.md:70]: Same comment as before for "CHIP lwip version" --> repos does not provide the middleware.

File: docs/discussion/lwip_ipv6.md [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+    the api / sockets layers
+-   Audit CHIP code to ensure LL addresses are properly scoped to their netif in
+    all areas (DNS returned addresses especially)
+
+## Duplicate address detection
+
+The DAD in LwIP is actually implemented correctly right now, but there are
+routers that incorrectly implement multicast for IPv6 and send packets back to
+the sender. This triggers the LwIP DAD because it doesn’t check the source. In
+our previous products, we fixed this as the wifi layer as a filter, but it’s
```
Comment (tcarmelveilleux) [inline docs/discussion/lwip_ipv6.md:81]: Suggest better wording that "our previous products"

File: docs/discussion/lwip_ipv6.md [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+easy enough to add the fix into the LwIP layer. This would help implementers so
+they don’t all have to debug the same issues. Recommendation:
+
+-   Create an LwIP patch to check NS/NA packets for source and discard if they
+    originate from the same device. Upstream and add to CHIP.
+
+## IPv6 Ping
+
+LwIP will automatically respond to pings, but has no built-in way to send them.
```
Comment (tcarmelveilleux) [inline docs/discussion/lwip_ipv6.md:90]: Not sure why we need to implement ICMPv6 ping for Matter, although it is a good idea. It is not required by the spec to send ICMPv6 pings.

**Review by tcarmelveilleux — APPROVED** [review]

File: docs/discussion/lwip_ipv6.md (hunk 6 above)
Comment (cecille) [reply docs/discussion/lwip_ipv6.md:90]: We don't need it for chip, but given that this is probably one of the first times a lot of devs will be working exclusively with IPv6 - ping can be a really helpful tool. I'll add a comment clarifying that this is something we should consider as a tool for development, not as a requirement of the spec.

**Review by woody-apple — APPROVED** [review]

**Comment by woody-apple** [comment]
@saurabhst @Damian-Nordic @msandstedt ?

**Review by andy31415 — APPROVED** [review]

**Review by msandstedt — APPROVED** [review]

**Review by jelderton — APPROVED** [review]

**Comment by woody-apple** [comment]
@cecille Please restyle :)

**Comment by andy31415** [comment]
Restyle passes now, will merge as md changes should not affect CI compilation runs.

**Pull request merged**